### PR TITLE
[py] refactored `Proxy` object in `common/proxy.py` using custom descriptor.

### PR DESCRIPTION
**Pull request by sandeepsuryaprasad**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description
Moved all `Proxy` object `properties` to custom descriptor class.

### Motivation and Context
- By Moving all `getters` and `setters` to custom descriptors, multiple `properties` are being eliminated.
- Currently there are 10 pairs of `property` with 20 methods. All doing similar job of validating the proxy type and assigning a class attribute with some proxy, there is lot of code repetition.
- This is now completely optimised by moving all `properties` to descriptor class.
- This reduces maintenance effort since the entire logic of attribute assignment lies in `__set__` method of descriptor.
- Also, refactored `to_capabilites` method in `Proxy` class by eliminating multiple conditional statements.

### Types of changes

- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [] My change requires a change to the documentation.
- [] I have updated the documentation accordingly.
- [] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: py/selenium/webdriver/common/proxy.py [hunk 1]
```diff
@@ -69,12 +110,169 @@ class Proxy:
...
+    - Get
+        - `self.proxy_type`
+    - Set
+        - `self.proxy_type` = `value`
+
+    Parameters
+    ----------
+    `value`: `proxy type`
+    """
+    auto_detect = _ProxyTypeDescriptor("autodetect", ProxyType.AUTODETECT)
```
Comment (AutomatedTester) [inline py/selenium/webdriver/common/proxy.py]: Please have a space above these lines to make it easier to read

**Review by AutomatedTester — CHANGES_REQUESTED** [review]
Just the readability changes and I will merge once updated

File: py/selenium/webdriver/common/proxy.py (hunk 1 above)
Comment (sandeepsuryaprasad) [reply py/selenium/webdriver/common/proxy.py]: Done!

**Review by AutomatedTester — APPROVED** [review]

**Review by diemol — COMMENTED** [review]
@sandeepsuryaprasad please check the tests, they are failing.

**Comment by diemol** [comment]
Why was the PR closed?

**Comment by sandeepsuryaprasad** [comment]
> Why was the PR closed?

Fixed the issue that was causing UI tests to fail. Raising a new PR with the fix.

**Comment by titusfortner** [comment]
So, you can overwrite the code here with new code without raising a new PR. Ask us for help with the git magic to do these things.

The new PR is: #12286

**Pull request closed without merging**